### PR TITLE
Add pagination schema

### PR DESCRIPTION
**Pull request by alikins**
Add back pagination schema and params
    
Fixes https://github.com/ansible/galaxy-dev/issues/31

### CODE REVIEW

**Comment by alikins** [comment]
@cutwater rebased

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -247,3 +247,85 @@ paths:
...
+          example: '/namespaces/?offset=7&limit=1'
+        next:
+          type: string
+          description: 'Link to next page'
+          format: uri
+          example: '/namespaces/?offset=5&limit=1'
+          nullable: true
+        previous:
+          type: string
+          description: 'Link to next page'
```
Comment (chouseknecht) [inline openapi/openapi.yaml]: Description should be `Link to previous page`

File: openapi/openapi.yaml [hunk 2]
```diff
@@ -247,3 +247,85 @@ paths:
           description: The current User
         'default':
           description: Error
+components:
+  schemas:
+
+    PageCount:
```
Comment (cutwater) [inline openapi/openapi.yaml]: Is there a reason of extracting `PageCount` as a standalone schema if it's used only in `PageMeta` schema and it's short. I think it creates too many utility standalone schemas that are not reused anywhere.

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -247,3 +247,85 @@ paths:
...
+      properties:
+        meta:
+          $ref: '#/components/schemas/PageMeta'
+        links:
+          $ref: '#/components/schemas/PageLinks'
+      required:
+        - meta
+        - links
+
+    PageLinks:
```
Comment (chouseknecht) [inline openapi/openapi.yaml]: I agree that there's no need to extract *PageCount*. Wondering why we're extracting *PageLinks*. Can we combine all the properties into *PageInfo* and make this simpler.

Comment (cutwater) [reply openapi/openapi.yaml]: Having standalone PageLinks might be reasonable, depending on how referencing this schema from endpoints.

File: openapi/openapi.yaml (hunk 2 above)
Comment (alikins) [reply openapi/openapi.yaml]: Yeah, it's def can be inlined.

File: openapi/openapi.yaml (hunk 3 above)
Comment (alikins) [reply openapi/openapi.yaml]: The JSON API spec that the insights pagination is based on considers 'links' it's own object (https://jsonapi.org/format/#document-links) so following that convention.

It is also how other insights apps model it:
https://github.com/RedHatInsights/vulnerability-engine/blob/master/manager.spec.yaml#L611
https://github.com/RedHatInsights/insights-rbac/blob/master/docs/source/specs/openapi.json#L1214

File: openapi/openapi.yaml (hunk 1 above)
Comment (alikins) [reply openapi/openapi.yaml]: updated

File: openapi/openapi.yaml (hunk 2 above)
Comment (alikins) [reply openapi/openapi.yaml]: inlined

**Comment by alikins** [comment]
@cutwater @chouseknecht

**Pull request merged**